### ISSUE TITLE
Container crashes when using "docker run" and graphite (Error: getaddrinfo EAI_AGAIN graphite)

### ISSUE THREAD

**Issue by alexdangelo** [sitespeedio/sitespeed.io#1798]
Expectation: using the docker container to run sitespeed and graphite runs without errors
Actual: crashes ("_ERROR: Error: getaddrinfo EAI_AGAIN graphite_")

Setup: Use the docker compose file to run Sitespeed + Grafana per https://www.sitespeed.io/documentation/sitespeed.io/performance-dashboard/#docker-compose-file
Windows 10 (build 15063.674)

**Good case**:
```
c:\docker>docker run --privileged --shm-size=1g --rm -v c:\docker:/sitespeed.io sitespeedio/sitespeed.io https://www.payscale.com/research -n 1
```
Output:
```
Google Chrome 60.0.3112.78
Mozilla Firefox 54.0
[2017-11-16 23:17:45] INFO: Versions OS: linux 4.9.49-moby nodejs: v6.11.1 sitespeed.io: 5.6.5 browsertime: 1.9.4 coach: 0.36.0
[2017-11-16 23:17:46] INFO: Starting chrome for analysing https://www.payscale.com/research 3 time(s)
[2017-11-16 23:17:46] INFO: Testing url https://www.payscale.com/research run 1
[2017-11-16 23:17:53] INFO: Testing url https://www.payscale.com/research run 2
[2017-11-16 23:17:59] INFO: Testing url https://www.payscale.com/research run 3
[2017-11-16 23:18:05] INFO: 96 requests, 1031.77 kb, backEndTime: 794ms (±24.15ms), firstPaint: 1.44s (±39.81ms), DOMContentLoaded: 2.11s (±31.28ms), Load: 3.05s (±130.51ms), rumSpeedIndex: 1443 (±38.59) (3 runs)
[2017-11-16 23:18:07] INFO: HTML stored in /sitespeed.io/sitespeed-result/www.payscale.com/2017-11-16-23-17-45
[2017-11-16 23:18:07] INFO: Finished analysing https://www.payscale.com/research
```

**Bad case**:
```
c:\docker>docker run --privileged --shm-size=1g --rm -v c:\docker:/sitespeed.io sitespeedio/sitespeed.io https://www.payscale.com/research --graphite.host=graphite --webpagetest.key WPT_KEY_HERE --webpagetest.runs 1 --video --speedIndex --firstParty ".*payscale.*" -n 1
```
Output:
```
Google Chrome 60.0.3112.78
Mozilla Firefox 54.0
[2017-11-16 23:19:22] INFO: Versions OS: linux 4.9.49-moby nodejs: v6.11.1 sitespeed.io: 5.6.5 browsertime: 1.9.4 coach: 0.36.0
[2017-11-16 23:19:22] INFO: Sending url https://www.payscale.com/research to test on https://www.webpagetest.org
[2017-11-16 23:19:22] INFO: Starting chrome for analysing https://www.payscale.com/research 1 time(s)
[2017-11-16 23:19:22] INFO: Testing url https://www.payscale.com/research run 1
[2017-11-16 23:19:38] INFO: 96 requests, 1030.68 kb, backEndTime: 696ms, firstPaint: 1.40s, firstVisualChange: 1.43s, DOMContentLoaded: 2.09s, Load: 2.86s, speedIndex: 1861, visualComplete85: 2.38s, lastVisualChange: 5.20s, rumSpeedIndex: 1402
[2017-11-16 23:19:39] ERROR: Error: getaddrinfo EAI_AGAIN graphite:2003
    at Object.exports._errnoException (util.js:1018:11)
    at errnoException (dns.js:33:15)
    at GetAddrInfoReqWrap.onlookup [as oncomplete] (dns.js:76:26)
From previous event:
    at GraphiteSender.send (/usr/src/app/lib/plugins/graphite/sender.js:16:12)
    at Object.processMessage (/usr/src/app/lib/plugins/graphite/index.js:66:26)
    at queue.process.message (/usr/src/app/lib/support/queueHandler.js:192:32)
    at drainItem (/usr/src/app/node_modules/concurrent-queue/index.js:96:21)
    at Immediate.drain (/usr/src/app/node_modules/concurrent-queue/index.js:77:84)
    at runCallback (timers.js:672:20)
    at tryOnImmediate (timers.js:645:5)
    at processImmediate [as _immediateCallback] (timers.js:617:5)
    at Socket.Readable.on (_stream_readable.js:687:35)
    at Promise (/usr/src/app/lib/plugins/graphite/sender.js:22:14)
    at GraphiteSender.send (/usr/src/app/lib/plugins/graphite/sender.js:16:12)
    at Object.processMessage (/usr/src/app/lib/plugins/graphite/index.js:66:26)
    at queue.process.message (/usr/src/app/lib/support/queueHandler.js:192:32)
    at drainItem (/usr/src/app/node_modules/concurrent-queue/index.js:96:21)
    at Immediate.drain (/usr/src/app/node_modules/concurrent-queue/index.js:77:84)
    at runCallback (timers.js:672:20)
    at tryOnImmediate (timers.js:645:5)
    at processImmediate [as _immediateCallback] (timers.js:617:5)
[2017-11-16 23:19:39] ERROR: Error: getaddrinfo EAI_AGAIN graphite:2003
    at Object.exports._errnoException (util.js:1018:11)
    at errnoException (dns.js:33:15)
    at GetAddrInfoReqWrap.onlookup [as oncomplete] (dns.js:76:26)
From previous event:
    at GraphiteSender.send (/usr/src/app/lib/plugins/graphite/sender.js:16:12)
    at Object.processMessage (/usr/src/app/lib/plugins/graphite/index.js:66:26)
    at queue.process.message (/usr/src/app/lib/support/queueHandler.js:192:32)
    at drainItem (/usr/src/app/node_modules/concurrent-queue/index.js:96:21)
    at Immediate.drain (/usr/src/app/node_modules/concurrent-queue/index.js:77:84)
    at runCallback (timers.js:672:20)
    at tryOnImmediate (timers.js:645:5)
    at processImmediate [as _immediateCallback] (timers.js:617:5)
    at Socket.Readable.on (_stream_readable.js:687:35)
    at Promise (/usr/src/app/lib/plugins/graphite/sender.js:22:14)
    at GraphiteSender.send (/usr/src/app/lib/plugins/graphite/sender.js:16:12)
    at Object.processMessage (/usr/src/app/lib/plugins/graphite/index.js:66:26)
    at queue.process.message (/usr/src/app/lib/support/queueHandler.js:192:32)
    at drainItem (/usr/src/app/node_modules/concurrent-queue/index.js:96:21)
    at Immediate.drain (/usr/src/app/node_modules/concurrent-queue/index.js:77:84)
    at runCallback (timers.js:672:20)
    at tryOnImmediate (timers.js:645:5)
    at processImmediate [as _immediateCallback] (timers.js:617:5)
[2017-11-16 23:19:39] ERROR: Error: getaddrinfo EAI_AGAIN graphite:2003
    at Object.exports._errnoException (util.js:1018:11)
    at errnoException (dns.js:33:15)
    at GetAddrInfoReqWrap.onlookup [as oncomplete] (dns.js:76:26)
From previous event:
    at GraphiteSender.send (/usr/src/app/lib/plugins/graphite/sender.js:16:12)
    at Object.processMessage (/usr/src/app/lib/plugins/graphite/index.js:66:26)
    at queue.process.message (/usr/src/app/lib/support/queueHandler.js:192:32)
    at drainItem (/usr/src/app/node_modules/concurrent-queue/index.js:96:21)
    at Immediate.drain (/usr/src/app/node_modules/concurrent-queue/index.js:77:84)
    at runCallback (timers.js:672:20)
    at tryOnImmediate (timers.js:645:5)
    at processImmediate [as _immediateCallback] (timers.js:617:5)
    at Socket.Readable.on (_stream_readable.js:687:35)
    at Promise (/usr/src/app/lib/plugins/graphite/sender.js:22:14)
    at GraphiteSender.send (/usr/src/app/lib/plugins/graphite/sender.js:16:12)
    at Object.processMessage (/usr/src/app/lib/plugins/graphite/index.js:66:26)
    at queue.process.message (/usr/src/app/lib/support/queueHandler.js:192:32)
    at drainItem (/usr/src/app/node_modules/concurrent-queue/index.js:96:21)
    at Immediate.drain (/usr/src/app/node_modules/concurrent-queue/index.js:77:84)
    at runCallback (timers.js:672:20)
    at tryOnImmediate (timers.js:645:5)
    at processImmediate [as _immediateCallback] (timers.js:617:5)
[2017-11-16 23:19:42] INFO: Got https://www.payscale.com/research analysed with id 171116_GP_1bd7b3f989b118c8f0ebb832068a1ab9 from https://www.webpagetest.org
[2017-11-16 23:19:43] ERROR: Error: getaddrinfo EAI_AGAIN graphite:2003
    at Object.exports._errnoException (util.js:1018:11)
    at errnoException (dns.js:33:15)
    at GetAddrInfoReqWrap.onlookup [as oncomplete] (dns.js:76:26)
From previous event:
    at GraphiteSender.send (/usr/src/app/lib/plugins/graphite/sender.js:16:12)
    at Object.processMessage (/usr/src/app/lib/plugins/graphite/index.js:66:26)
    at queue.process.message (/usr/src/app/lib/support/queueHandler.js:192:32)
    at drainItem (/usr/src/app/node_modules/concurrent-queue/index.js:96:21)
    at Immediate.drain (/usr/src/app/node_modules/concurrent-queue/index.js:77:84)
    at runCallback (timers.js:672:20)
    at tryOnImmediate (timers.js:645:5)
    at processImmediate [as _immediateCallback] (timers.js:617:5)
    at Socket.Readable.on (_stream_readable.js:687:35)
    at Promise (/usr/src/app/lib/plugins/graphite/sender.js:22:14)
    at GraphiteSender.send (/usr/src/app/lib/plugins/graphite/sender.js:16:12)
    at Object.processMessage (/usr/src/app/lib/plugins/graphite/index.js:66:26)
    at queue.process.message (/usr/src/app/lib/support/queueHandler.js:192:32)
    at drainItem (/usr/src/app/node_modules/concurrent-queue/index.js:96:21)
    at Immediate.drain (/usr/src/app/node_modules/concurrent-queue/index.js:77:84)
    at runCallback (timers.js:672:20)
    at tryOnImmediate (timers.js:645:5)
    at processImmediate [as _immediateCallback] (timers.js:617:5)
[2017-11-16 23:19:43] ERROR: Error: getaddrinfo EAI_AGAIN graphite:2003
    at Object.exports._errnoException (util.js:1018:11)
    at errnoException (dns.js:33:15)
    at GetAddrInfoReqWrap.onlookup [as oncomplete] (dns.js:76:26)
From previous event:
    at GraphiteSender.send (/usr/src/app/lib/plugins/graphite/sender.js:16:12)
    at Object.processMessage (/usr/src/app/lib/plugins/graphite/index.js:66:26)
    at queue.process.message (/usr/src/app/lib/support/queueHandler.js:192:32)
    at drainItem (/usr/src/app/node_modules/concurrent-queue/index.js:96:21)
    at Immediate.drain (/usr/src/app/node_modules/concurrent-queue/index.js:77:84)
    at runCallback (timers.js:672:20)
    at tryOnImmediate (timers.js:645:5)
    at processImmediate [as _immediateCallback] (timers.js:617:5)
    at Socket.Readable.on (_stream_readable.js:687:35)
    at Promise (/usr/src/app/lib/plugins/graphite/sender.js:22:14)
    at GraphiteSender.send (/usr/src/app/lib/plugins/graphite/sender.js:16:12)
    at Object.processMessage (/usr/src/app/lib/plugins/graphite/index.js:66:26)
    at queue.process.message (/usr/src/app/lib/support/queueHandler.js:192:32)
    at drainItem (/usr/src/app/node_modules/concurrent-queue/index.js:96:21)
    at Immediate.drain (/usr/src/app/node_modules/concurrent-queue/index.js:77:84)
    at runCallback (timers.js:672:20)
    at tryOnImmediate (timers.js:645:5)
    at processImmediate [as _immediateCallback] (timers.js:617:5)
[2017-11-16 23:19:43] ERROR: Error: getaddrinfo EAI_AGAIN graphite:2003
    at Object.exports._errnoException (util.js:1018:11)
    at errnoException (dns.js:33:15)
    at GetAddrInfoReqWrap.onlookup [as oncomplete] (dns.js:76:26)
From previous event:
    at GraphiteSender.send (/usr/src/app/lib/plugins/graphite/sender.js:16:12)
    at Object.processMessage (/usr/src/app/lib/plugins/graphite/index.js:66:26)
    at queue.process.message (/usr/src/app/lib/support/queueHandler.js:192:32)
    at drainItem (/usr/src/app/node_modules/concurrent-queue/index.js:96:21)
    at Immediate.drain (/usr/src/app/node_modules/concurrent-queue/index.js:77:84)
    at runCallback (timers.js:672:20)
    at tryOnImmediate (timers.js:645:5)
    at processImmediate [as _immediateCallback] (timers.js:617:5)
    at Socket.Readable.on (_stream_readable.js:687:35)
    at Promise (/usr/src/app/lib/plugins/graphite/sender.js:22:14)
    at GraphiteSender.send (/usr/src/app/lib/plugins/graphite/sender.js:16:12)
    at Object.processMessage (/usr/src/app/lib/plugins/graphite/index.js:66:26)
    at queue.process.message (/usr/src/app/lib/support/queueHandler.js:192:32)
    at drainItem (/usr/src/app/node_modules/concurrent-queue/index.js:96:21)
    at Immediate.drain (/usr/src/app/node_modules/concurrent-queue/index.js:77:84)
    at runCallback (timers.js:672:20)
    at tryOnImmediate (timers.js:645:5)
    at processImmediate [as _immediateCallback] (timers.js:617:5)
[2017-11-16 23:19:43] ERROR: Error: getaddrinfo EAI_AGAIN graphite:2003
    at Object.exports._errnoException (util.js:1018:11)
    at errnoException (dns.js:33:15)
    at GetAddrInfoReqWrap.onlookup [as oncomplete] (dns.js:76:26)
From previous event:
    at GraphiteSender.send (/usr/src/app/lib/plugins/graphite/sender.js:16:12)
    at Object.processMessage (/usr/src/app/lib/plugins/graphite/index.js:66:26)
    at queue.process.message (/usr/src/app/lib/support/queueHandler.js:192:32)
    at drainItem (/usr/src/app/node_modules/concurrent-queue/index.js:96:21)
    at Immediate.drain (/usr/src/app/node_modules/concurrent-queue/index.js:77:84)
    at runCallback (timers.js:672:20)
    at tryOnImmediate (timers.js:645:5)
    at processImmediate [as _immediateCallback] (timers.js:617:5)
    at Socket.Readable.on (_stream_readable.js:687:35)
    at Promise (/usr/src/app/lib/plugins/graphite/sender.js:22:14)
    at GraphiteSender.send (/usr/src/app/lib/plugins/graphite/sender.js:16:12)
    at Object.processMessage (/usr/src/app/lib/plugins/graphite/index.js:66:26)
    at queue.process.message (/usr/src/app/lib/support/queueHandler.js:192:32)
    at drainItem (/usr/src/app/node_modules/concurrent-queue/index.js:96:21)
    at Immediate.drain (/usr/src/app/node_modules/concurrent-queue/index.js:77:84)
    at runCallback (timers.js:672:20)
    at tryOnImmediate (timers.js:645:5)
    at processImmediate [as _immediateCallback] (timers.js:617:5)
[2017-11-16 23:19:43] ERROR: Error: getaddrinfo EAI_AGAIN graphite:2003
    at Object.exports._errnoException (util.js:1018:11)
    at errnoException (dns.js:33:15)
    at GetAddrInfoReqWrap.onlookup [as oncomplete] (dns.js:76:26)
From previous event:
    at GraphiteSender.send (/usr/src/app/lib/plugins/graphite/sender.js:16:12)
    at Object.processMessage (/usr/src/app/lib/plugins/graphite/index.js:66:26)
    at queue.process.message (/usr/src/app/lib/support/queueHandler.js:192:32)
    at drainItem (/usr/src/app/node_modules/concurrent-queue/index.js:96:21)
    at Immediate.drain (/usr/src/app/node_modules/concurrent-queue/index.js:77:84)
    at runCallback (timers.js:672:20)
    at tryOnImmediate (timers.js:645:5)
    at processImmediate [as _immediateCallback] (timers.js:617:5)
    at Socket.Readable.on (_stream_readable.js:687:35)
    at Promise (/usr/src/app/lib/plugins/graphite/sender.js:22:14)
    at GraphiteSender.send (/usr/src/app/lib/plugins/graphite/sender.js:16:12)
    at Object.processMessage (/usr/src/app/lib/plugins/graphite/index.js:66:26)
    at queue.process.message (/usr/src/app/lib/support/queueHandler.js:192:32)
    at drainItem (/usr/src/app/node_modules/concurrent-queue/index.js:96:21)
    at Immediate.drain (/usr/src/app/node_modules/concurrent-queue/index.js:77:84)
    at runCallback (timers.js:672:20)
    at tryOnImmediate (timers.js:645:5)
    at processImmediate [as _immediateCallback] (timers.js:617:5)
[2017-11-16 23:19:43] ERROR: Error: getaddrinfo EAI_AGAIN graphite:2003
    at Object.exports._errnoException (util.js:1018:11)
    at errnoException (dns.js:33:15)
    at GetAddrInfoReqWrap.onlookup [as oncomplete] (dns.js:76:26)
From previous event:
    at GraphiteSender.send (/usr/src/app/lib/plugins/graphite/sender.js:16:12)
    at Object.processMessage (/usr/src/app/lib/plugins/graphite/index.js:66:26)
    at queue.process.message (/usr/src/app/lib/support/queueHandler.js:192:32)
    at drainItem (/usr/src/app/node_modules/concurrent-queue/index.js:96:21)
    at Immediate.drain (/usr/src/app/node_modules/concurrent-queue/index.js:77:84)
    at runCallback (timers.js:672:20)
    at tryOnImmediate (timers.js:645:5)
    at processImmediate [as _immediateCallback] (timers.js:617:5)
    at Socket.Readable.on (_stream_readable.js:687:35)
    at Promise (/usr/src/app/lib/plugins/graphite/sender.js:22:14)
    at GraphiteSender.send (/usr/src/app/lib/plugins/graphite/sender.js:16:12)
    at Object.processMessage (/usr/src/app/lib/plugins/graphite/index.js:66:26)
    at queue.process.message (/usr/src/app/lib/support/queueHandler.js:192:32)
    at drainItem (/usr/src/app/node_modules/concurrent-queue/index.js:96:21)
    at Immediate.drain (/usr/src/app/node_modules/concurrent-queue/index.js:77:84)
    at runCallback (timers.js:672:20)
    at tryOnImmediate (timers.js:645:5)
    at processImmediate [as _immediateCallback] (timers.js:617:5)
[2017-11-16 23:19:45] INFO: HTML stored in /sitespeed.io/sitespeed-result/www.payscale.com/2017-11-16-23-19-22
[2017-11-16 23:19:45] INFO: Finished analysing https://www.payscale.com/research
```

**Comment by soulgalore** [comment]
If you use "graphite" as you do, you need to share the same network, checkout @beenanner answer here: https://github.com/sitespeedio/sitespeed.io/issues/1792#issuecomment-343811214

**Comment by alexdangelo** [comment]
Thanks for the pointer. I haven't been able to get the "docker run" version to work even with the example in the thread but I'm sure it's how I'm passing graphite params to docker and not an issue with Sitespeed itself. (using `docker run` for Sitespeed works great when the graphite params aren't included). I'll use the docker-compose approach when using graphite instead.

(Here are my notes of what I tried anyway)
I ran
```
C:\docker>docker run --network docker_default --privileged --shm-size=1g --rm -v /c/docker:/sitespeed.io sitespeedio/sitespeed.io urls.txt --graphite.host=graphite --webpagetest.key WPT_KEY_HERE --webpagetest.runs 1 --video --speedIndex --firstParty ".*payscale.*" -n 1
```
and the parameter for `network` comes from running 
```
C:\docker>docker network ls
NETWORK ID          NAME                DRIVER              SCOPE
e1931f022973        bridge              bridge              local
df5e1c60aaf1        docker_default      bridge              local
8acc3204c2cc        host                host                local
74af97853f53        none                null                local
```

**Comment by alexdangelo** [comment]
Turns out the solution I tried in https://github.com/sitespeedio/sitespeed.io/issues/1798#issuecomment-345394677 worked! 

I was monitoring results in this dashboard: http://localhost:3000/dashboard/db/page-summary

Why didn't I see the results appear in Grafana before, even after pressing the Grafana refresh button?

The answer is that  pressing the refresh button in Grafana (shown below) didn't seem to pick up that there was a new Page in the page dropdown. 
![image](https://user-images.githubusercontent.com/1562950/32974623-6cc09be6-cbb4-11e7-8128-2659ef481f5d.png)
Instead, reloading the Grafana browser tab, or switching dashboard refreshed what appeared in the Pages dropdown.
![image](https://user-images.githubusercontent.com/1562950/32974692-0913606e-cbb5-11e7-8757-ea2496f9eb01.png)
